service/storage_proxy.cc

```diff
@@ -6545,7 +6545,37 @@ db::hints::manager& storage_proxy::hints_manager_for(db::write_type type) {
 }
 
 future<> storage_proxy::truncate_blocking(sstring keyspace, sstring cfname, std::optional<std::chrono::milliseconds> timeout_in_ms) {
-    return remote().send_truncate_blocking(std::move(keyspace), std::move(cfname), timeout_in_ms);
+    locator::effective_replication_map_ptr erm;
+    const auto& ks = _db.local().find_keyspace(keyspace);
+    if (ks.get_replication_strategy().uses_tablets()) {
+        slogger.debug("Running TRUNCATE TABLE (with tablets) on {}.{}", keyspace, cfname);
+        co_await utils::get_local_injector().inject("truncate_with_tablets_wait_before", [] (auto& handler) {
+            slogger.debug("truncate_with_tablets_wait_before: start");
+            return handler.wait_for_message(db::timeout_clock::now() + std::chrono::minutes(2));
```

Comment:
This adds a long wait, which suggests that the test may have a long run time. What kind of synchronization do you want to achieve with this wait? Maybe there is a better way.

Reply:
I use the message `truncate_with_tablets_wait_before` to start the truncate process while tablet migration is executing, to then check if truncate really waited for migration to complete before starting.
The 2 minute wait is just a (very conservative) timeout. I set it high to make sure to avoid flaky tests by setting it too low. I could lower it to 30 seconds, but I don't think we should go lower than that. Even leaving it at 2 minutes, and adding a comment describing why it is this high (flaky test) seems to me like a better solution then pushing our luck by setting it loo low. Do you insist I lower this?

Reply:
> Do you insist I lower this?

No, it's fine. Thanks for the clarification.